components/service/nimbus/src/main/java/mozilla/components/service/nimbus/Nimbus.kt

```diff
@@ -173,29 +222,81 @@ class Nimbus(
         nimbus.getExperimentBranch(experimentId)
 
     override fun updateExperiments() {
-        scope.launch {
-            try {
-                nimbus.updateExperiments()
-
-                // Get the experiments to record in telemetry
-                nimbus.getActiveExperiments().let {
-                    if (it.any()) {
-                        recordExperimentTelemetry(it)
-                        // The current plan is to have the nimbus-sdk updateExperiments() function
-                        // return a diff of the experiments that have been received, at which point we
-                        // can emit the appropriate telemetry events and notify observers of just the
-                        // diff
-                        notifyObservers { onUpdatesApplied(it) }
-                    }
-                }
-            } catch (e: ErrorException.RequestError) {
-                logger.info("Error fetching experiments from endpoint: $e")
-            } catch (e: ErrorException.InvalidExperimentResponse) {
-                logger.info("Invalid experiment response: $e")
+        networkScope.launch {
+            fetchExperimentsOnThisThread()
+            applyPendingExperimentsOnThisThread()
+        }
+    }
+
+    override fun fetchExperiments() {
+        networkScope.launch {
+            fetchExperimentsOnThisThread()
+        }
+    }
+
+    @WorkerThread
+    @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+    private fun fetchExperimentsOnThisThread() {
+        try {
+            nimbus.fetchExperiments()
+            notifyObservers { onExperimentsFetched() }
+        } catch (e: ErrorException.RequestError) {
+            logger.info("Error fetching experiments from endpoint: $e")
+        }
+    }
+
+    override fun applyPendingExperiments() {
+        dbScope.launch {
+            applyPendingExperimentsOnThisThread()
+        }
+    }
+
+    @WorkerThread
+    @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+    private fun applyPendingExperimentsOnThisThread() {
+        try {
+            nimbus.applyPendingExperiments()
+            // Get the experiments to record in telemetry
+            postEnrolmentCalculation()
+        } catch (e: ErrorException.InvalidExperimentFormat) {
+            logger.info("Invalid experiment format: $e")
+        }
+    }
+
+    private fun postEnrolmentCalculation() {
+        nimbus.getActiveExperiments().let {
+            if (it.any()) {
+                recordExperimentTelemetry(it)
+                // The current plan is to have the nimbus-sdk updateExperiments() function
+                // return a diff of the experiments that have been received, at which point we
+                // can emit the appropriate telemetry events and notify observers of just the
+                // diff
```

Comment:
@travis79 Is this comment still useful?

Reply:
There is still an open issue for this: [SDK-6](https://jira.mozilla.com/browse/SDK-6).  The comment can probably be removed if it is distracting and since we have an issue that captures this info until it is implemented.